### DISCUSSION TITLE
Is there a way to backup the current config?

### DISCUSSION THREAD

**Discussion by TerryYoung61** [RaspAP/raspap-webgui#1243] [General]
Is there a way to backup (and restore) the current configuration, for if something goes wrong?
It's a pretty standard feature on most other AP's / router systems and was wondering if it's possible under the current version, or if it could maybe be implemented?
I just spent ages trying to get it working, and know that just one thing clicked wrong can break the system.
Thanks a lot,  and love the software (much less complicated than OpenWRT)

**Comment by billz** [comment]
Good question. RaspAP [backs up your existing configuration](https://github.com/RaspAP/raspap-webgui/blob/4c23b453757cdfa665dc39a91e763e9e9b0fec06/installers/common.sh#L496) to `/etc/raspap/backups` so they may be restored by the uninstaller. These are your system configs before running the installer, so they don't reflect subsequent changes made by RaspAP.

Changes made via the UI are written to `dhcpcd.conf`, `hostapd.conf` and one or more files in `/etc/dnsmasq.d/`. There are also  OpenVPN and WireGuard configs (optional), plus [RaspAP's own configuration file](https://github.com/RaspAP/raspap-webgui/blob/master/config/config.php) and user auth file. So quite a few backup items to manage in various locations.

There's no backup/restore facility at the moment, but I agree it would be a nice-to-have feature.
